src/testing.py

```diff
@@ -75,7 +75,10 @@ def setlimits():
   def __call__(self, test_file):
     """Execute a single test."""
     basename = os.path.basename(test_file)
-    outfile = self.outname_ctor(self.outdir, test_file) if self.outdir else ''
+    if self.outdir:
+      outfile = self.outname_ctor(self.outdir, test_file, self.extras)
+    else:
+      outfile = None
```

Comment:
Well the status quo is that execute_files seems to handle have no outfile at all.  maybe it doesn't need that anymore and we can remove support, but that would be a separate change I think